go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/buildkite/go-pipeline
 
 go 1.21.0
```

Comment:
If the agent uses this, does force the agent to be on 1.21 as well? And usually I see it as
```suggestion
go 1.21
```

Reply:
I guess technically not: https://go.dev/doc/modules/gomod-ref#go-notes it's only after go 1.21 that it's enforced.

Reply:
"Before Go 1.21, the directive was advisory only; now it is a mandatory requirement: Go toolchains refuse to use modules declaring newer Go versions."

And yeah, `1.21.0` might choke Go 1.20's go.mod parser.

Reply:
jinx!